src/rust/catnap/queue.rs

```diff
@@ -293,6 +369,32 @@ impl SharedCatnapQueue {
         }
     }
 
+    /// Generic function for spawning a control-path coroutine on [self].
+    fn do_generic_sync_control_path_call<F>(
+        &mut self,
+        coroutine_constructor: F,
+        yielder: Yielder,
+    ) -> Result<TaskHandle, Fail>
+    where
+        F: FnOnce(Yielder) -> Result<TaskHandle, Fail>,
+    {
+        // Spawn coroutine.
```

Comment:
With the new name `coroutine_constructor`, this comment probably needs to be revisited.

Reply:
Done